### ISSUE TITLE
[MWRAPPER-22] maven-wrapper on windows

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-wrapper#196]
**[James Z.M. Gao](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=gzm55)** opened **[MWRAPPER-22](https://issues.apache.org/jira/browse/MWRAPPER-22?redirect=false)** and commented

new maven-wrapper inherit 2 bugs for windows batch scripts:

1. cannot read the config ".mvn\wrapper\maven-wrapper.properties" correctly when the full path has space.
2. -eof style of cmd scripts are not CRLF- (This is fixed by assembly-plugin `lineEnding` settings)

---

**Affects:** 0.5.6

**Remote Links:**
- [GitHub Pull Request #353
](https://github.com/apache/maven/pull/353)

**Comment by jira-importer** [comment]
**[Robert Scholte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rfscholte)** commented

#1 suprises me. There are integration tests for all 3 wrapper distributions, they all run fine on our Jenkins: https://builds.apache.org/job/maven-box/job/maven/job/master/
#2 is IMO not an issue. Even though git changes the EOL style, in the end the assembly plugin will fix this when combining all script fragments.

**Comment by jira-importer** [comment]
**[James Z.M. Gao](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=gzm55)** commented

#2 ok, see the lineEnding settings in assembly bin.xml, i will revert this part; but the gitattributes file is still out of date, we should remove apache-maven/src/assembly/maven/bin/.gitattributes

**Comment by jira-importer** [comment]
**[James Z.M. Gao](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=gzm55)** commented

#1 when reading files with spaces in the full path, we need `usebackq` option, like in [init.cmd](https://github.com/apache/maven/blob/master/apache-maven/src/assembly/shared/init.cmd#L93), and this passes the IT tests on [travis](https://travis-ci.org/github/gzm55/maven-wrapper/builds/622324321).

**Comment by jira-importer** [comment]
**[Robert Scholte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rfscholte)** commented

Let me see if I can run the ITs on a path containing spaces first

**Comment by jira-importer** [comment]
**[Robert Scholte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rfscholte)** commented

After adding a space to a folder, I only see failures with the source type. The javac and java commands are missing proper quotes, so I'll fix that.
Can you share a failing build with the current scripts?

**Comment by jira-importer** [comment]
**[James Z.M. Gao](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=gzm55)** commented

> for /f "tokens=1,2 delims==" %A in ("a.txt")
> This will parse the literal string "a.txt", the spaces in the string are ok.

> for /f "usebackq tokens=1,2 delims==" %A in ("a.txt")
> This will parse the content in file a.txt, spaces in the path are ok.

> for /f "usebackq tokens=1,2 delims==" %A in (a.txt)
> This will parse the content in file a.txt, the path must not contain spaces.

example codes:

```none
C:\Users\james>more a.txt
k=v

C:\Users\james>for /f "tokens=1,2 delims==" %A in ("a.txt") do (echo %A to %B)

C:\Users\james>(echo a.txt to  )
a.txt to

C:\Users\james>for /f "usebackq tokens=1,2 delims==" %A in ("a.txt") do (echo %A to %B)

C:\Users\james>(echo k to v  )
k to v
```

To fail the current implementation, use a different download url in the wrapper.propertites file

**Comment by jira-importer** [comment]
**[Robert Scholte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rfscholte)** commented

actually, it looks more like we can remove this piece of code too.
Unlike the takari-maven-plugin, I've decided to say that the scripts and maven-wrapper.jar belong to the same release.
You cannot define the wrapperUrl anymore, the distributionUrl in the `wrapper.properties` is only thing used, and only by the Java code.

**Comment by jira-importer** [comment]
**[Robert Scholte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rfscholte)** commented

Fixed in [66f795476bc7c9f5c21f4a337fbdc1c39c7fd0c6](https://gitbox.apache.org/repos/asf?p=maven.git;a=commit;h=66f795476bc7c9f5c21f4a337fbdc1c39c7fd0c6)
Related ITs adjusted in [764176dbedd91cb27d4a3991958c92bb178d2836](https://gitbox.apache.org/repos/asf?p=maven-integration-testing.git;a=commit;h=764176dbedd91cb27d4a3991958c92bb178d2836)

**Comment by jira-importer** [comment]
**[Hervé Boutemy](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=hboutemy)** commented

PR reapplied in https://github.com/apache/maven-wrapper/commit/2873019f2c0d92c0da1f86cdd6cbbc10a44b6e0d

thank you
